rts/motoko-rts/src/alloc.rs

```diff
@@ -11,14 +11,14 @@ pub use alloc_impl::alloc_words;
 #[cfg(feature = "gc")]
 pub(crate) use alloc_impl::grow_memory;
 
+use crate::constants::WASM_HEAP_SIZE;
 use crate::rts_trap_with;
 use crate::types::{size_of, Array, Blob, Bytes, SkewedPtr, Words, TAG_ARRAY, TAG_BLOB};
 
 #[no_mangle]
 pub unsafe extern "C" fn alloc_array(len: u32) -> SkewedPtr {
     // Array payload should not be larger than half of the memory
```

Comment:
I wish we documented why this is the case. I believe this is because the copying GC will want to copy it and overflow the heap if this is 2GiB, but I don't know if there are other reasons.